### ISSUE TITLE
Accessibility issues on Settings > Payments screen

### ISSUE THREAD

**Issue by mcliwanow** [woothemes/woocommerce#60686]
When checking the latest changes on Settings > Payments screen (we were mostly focused on changes we made to the label in the "other payment options" section) on the `/wp-admin/admin.php?page=wc-settings&tab=checkout&from=PAYMENTS_MENU_ITEM` screen, @kokkieh and I noticed a couple of accessibility issues, which we thought were worth sharing:

1. Expand/Collapse icon in the "other payment options" section is not focusable with the keyboard. Likely wrapping it in a <button> and adding a screen reader text would be enough 

   <img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/9b2d2a7f-c9b6-43e7-aeed-06a872c3ce3f/cf364fa9-fea1-4935-b70c-1dcdf133ccff?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi85YjJkMmE3Zi1jOWI2LTQzZTctYWVlZC0wNmE4NzJjM2NlM2YvY2YzNjRmYTktZmVhMS00OTM1LWI3MGMtMWRjZGYxMzNjY2ZmIiwiaWF0IjoxNzU2NzE5NjUzLCJleHAiOjMzMzI3Mjc5NjUzfQ.ZXEiuzBi7rF7h5wwTQcWuI_3GjvuOY11kEtPjB306EQ " alt="image.png" width="2051" data-linear-height="135" />
2. When I open tooltips with a keyboard, I cannot navigate outside of them. It's good that focus is trapped within them, but it would be good if they closed when the Escape button is clicked.

<img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/76501713-480e-4809-a5f4-5d3fe76e3e9d/4c0667ae-25ef-480a-b93f-031707b229a1?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi83NjUwMTcxMy00ODBlLTQ4MDktYTVmNC01ZDNmZTc2ZTNlOWQvNGMwNjY3YWUtMjVlZi00ODBhLWI5M2YtMDMxNzA3YjIyOWExIiwiaWF0IjoxNzU2NzE5NjUzLCJleHAiOjMzMzI3Mjc5NjUzfQ.4JXv9xKnTglJcen5ai862EblskKaWjysUJLKXYZ36Gw " alt="Screenshot 2025-09-01 at 11.28.43.png" width="809" data-linear-height="218" />

3. This one might be worth a separate task, as I'm not sure if it's an issue with some core control or how it's used on the screen 🤔  Anyway, the "Business location" dropdown is also not really usable with keyboard - focus state is not clear and I cannot select countries with keyboard.

   <img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/26b7ffb0-6157-4907-ab04-7710ce6b665e/90e3679c-9e69-4e78-892c-aced2850fe8c?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi8yNmI3ZmZiMC02MTU3LTQ5MDctYWIwNC03NzEwY2U2YjY2NWUvOTBlMzY3OWMtOWU2OS00ZTc4LTg5MmMtYWNlZDI4NTBmZThjIiwiaWF0IjoxNzU2NzE5NjUzLCJleHAiOjMzMzI3Mjc5NjUzfQ.OkkIx8t6AIt8pao3Jx64Th3EClRVDTFa84ESpKZ4tmM " alt="image.png" width="1530" data-linear-height="136" />

@vlad.olaru, I think Moltres team manages this screen, so I took the liberty of assigning you to the issue. Let me know if I should tag similar issues differently in the future 🙏

**Comment by vladolaru** [comment]
Thank you so much for taking the time to share this @mcliwanow! Yes, you did good by assigning it to me.
